pkg/kubelet/kubelet.go

```diff
@@ -1926,6 +1966,23 @@ func (kl *Kubelet) canAdmitPod(pods []*v1.Pod, pod *v1.Pod) (bool, string, strin
 	// TODO: move out of disk check into a pod admitter
 	// TODO: out of resource eviction should have a pod admitter call-out
 	attrs := &lifecycle.PodAdmitAttributes{Pod: pod, OtherPods: pods}
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
```

Comment:
i need a little more time to understand the logic sequence here.

just recording my own notes so I can update this when I get through rest:

a mutex is being held outside this method to ensure no concurrent calls.
this flow is being invoked when a pod is first admitted (i.e. HandlePodAdditions and never from a pod update)
this is checking if the resource request can actually be allocated

Reply:
The mutation to use allocated resources for admitted is necessary in both cases because the desired resources may have changed and not accepted (Infeasible/Deferred).

canAdmitPod is called in handlePodResourcesResize only if we determine that pod resize is desired. Let me try illustrating with an example: 

Say we have 3 pods P1 (300m CPU) , P2 (100m CPU) , and P3 (500m CPU). Now P2 desires +100m CPU. This second canAdmitPod call leaves out the originally admitted pod P2 (100m CPU) that wants resize from the set of pods on the node, and tests if a brand new pod  P2 (200m CPU) would be admissible. If the answer is yes, then +100m CPU resize is admissible.

Does that clarify?

PS: I agree with always pull from checkpoint, let me take a closer look to see why I ever added that check.

Reply:
Fixed to always use checkpoint values. please review